Allow the Soul Forge to use any type of Will for Recipes.

VT-14 opened the pull request:
#1838

VT-14 committed:
Allow the Hellfire Forge to use any Demon Will Type

VT-14 committed:
Replaced a += with just =.

`soulsInGem` should never be non-zero when set to `getWill(type)`, so `+=` and `=` give the same result.

WayofTime commented on src/main/java/wayoftime/bloodmagic/tile/TileSoulForge.java:
```diff
@@ -131,7 +131,17 @@ public void tick()
+		for (EnumDemonWillType type : EnumDemonWillType.values())
+		{
+			soulsInGem = getWill(type);
+			if (soulsInGem > 0)
```
Should perform the check versus the minimum amount of will in the recipe, in the case of there being a gem with multiple aspects.

wrincewind replied:
A gem with multiple aspects? I'd never even considered that as a possibility... But it sounds like it'd be pretty cool.

VT-14 replied:
Adjusted by the latest commit.

VT-14 committed:
Support Potential Multi-Will Gems

Add support for potential Multi-Will Tartaric Gems.

Moved the Gem will level checks to after the valid recipe check.
